src/utils/layout.ts

```diff
@@ -44,28 +44,26 @@ export class LayoutController {
     updateDesiredPosition(commentId: number) {
         let annotation = this.commentAnnotations.get(commentId);
 
-        if (annotation.highlights.length === 0) {
-            return;
-        }
-
-        // Get average position
-        const sumOfPositions = annotation.highlights
-            .map(highlight => {
-                return (
-                    highlight.getBoundingClientRect().top +
-                    document.documentElement.scrollTop
-                );
-            })
-            .reduce((sum, position) => {
-                return sum + position;
-            });
-
-        const averagePosition = sumOfPositions / annotation.highlights.length;
+        this.commentDesiredPositions.set(commentId, annotation.getDesiredPosition() + OFFSET);
+    }
 
-        this.commentDesiredPositions.set(commentId, averagePosition + OFFSET);
+    refreshDesiredPositions() {
+        for (const commentId of this.commentAnnotations.keys()) {
+            this.updateDesiredPosition(commentId);
+          }
```

Comment:
**Nitpick:** Definitely okay with for of/in in Wagtail?

https://github.com/airbnb/javascript/blob/63098cbb6c05376dbefc9a91351f5727540c1ce1/packages/eslint-config-airbnb-base/rules/style.js#L339-L357

Reply:
Pretty sure, but switched out now anyway for peace of mind!